Automattic/jetpack
# Boost: Add cloud critical CSS generation feature (take 2)

Fixes 45-gh-Automattic/boost-hydra

#### Changes proposed in this Pull Request:
* Add a cloud CSS module based on critical CSS feature
* Make it possible to switch between cloud CSS and critical CSS, by defining `JETPACK_BOOST_CLOUD_CSS` as a boolean value.
* Refactor Critical CSS feature to move common classes to lib for sharing between the two modules.
* Add rest API endpoints to request and update cloud css

#### Jetpack product discussion
pc9hqz-fv-p2

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Start the docker development environment `jetpack docker up`.
* Add a tunnel `jetpack docker jt-up` and access the site through the tunnel.
* Add the development MU plugin for testing local setup (shared internally). 
* Start shield in development mode `npm run docker:dev`
* Start hydra in connected mode `npm run connected-docker`
* Try enabling or requesting regenerate Cloud CSS.
* You should see critical CSS in the front-end source.
